lxd/identities.go

```diff
@@ -1290,11 +1290,6 @@ func updateIdentity(authenticationMethod string) func(d *Daemon, r *http.Request
 			return response.SmartError(err)
 		}
 
-		// Only identities of type api.IdentityTypeCertificateClient may update their own certificate
-		if id.Type != api.IdentityTypeCertificateClient {
```

Comment:
@kadinsayani @markylaing does this backport look right (and the similar check removal below)?